### ISSUE TITLE
C#: DeserializeObject<object> is generated for 400/500 response when first response has no schema

### ISSUE THREAD

**Issue by calvin998** [Azure/autorest#2798]
This is the section of the swagger json:
`{
  "/api/v1/users/{userGuid}/emails/{email}": {
    "delete": {
      "tags": [        "Emails"      ],
      "summary": "Delete one email",
      "operationId": "Emails_DeleteEmail",
      "consumes": [],
      "produces": [],
      "parameters": [
        {
          "name": "userGuid",
          "in": "path",
          "description": "user guid",
          "required": true,
          "type": "string"
        },
        {
          "name": "email",
          "in": "path",
          "required": true,
          "type": "string"
        }
      ],
      "responses": {
        "204": {
          "description": "Email deleted"
        },
        "400": {
          "description": "Bad request. See return for details.",
          "schema": {
            "$ref": "#/definitions/ValidationErrorResponse"
          }
        },
        "500": {
          "description": "Server error. See return for details.",
          "schema": {
            "$ref": "#/definitions/ErrorResponse"
          }
        }
      }
    }
  }
}`

Please note that the first response has no schema. In the generated c# code, for 400 and 500 return types, this line was generated:

 _result.Body = Microsoft.Rest.Serialization.SafeJsonConvert.DeserializeObject<**object**>(_responseContent, Client.DeserializationSettings);

instead of <ValidationErrorResponse> or <ErrorResponse>. 

As soon as I add schema info to the first response, the last few response deserialization code will be generated correctly (correct data type is being used).

Currently this bug forces me to return a bool or integer for PUT/PATCH so that result can be deserialized to correct data type. However I really don't feel comfortable to set a return type (schema) for DELETE. Can this be fixed?

**Comment by fearthecowboy** [comment]
I've edited your comment so that it's readable. 

AutoRest (currently) assumes that all specified responses are "positive" responses, and that responses that should throw are bundled up as `"default"` .. (yeah, I know).

The good news, is that we're just finishing a feature to mark certain responses as `error` responses that will do exactly as you expect. 

The code should be getting merged in the next few days.

**Comment by calvin998** [comment]
To work with the the new feature (marking "error" response), do I have to change my swagger.json? My json is generated through swashbuckel from my web api controllers. There is very little room (if any) for me to control the generated json file.

**Comment by epet** [comment]
@fearthecowboy, has this been committed? We are experiencing the same thing.

**Comment by fearthecowboy** [comment]
@epet  it's not in the main line yet; it required changes to the client runtime; and there's a delay getting that published.

I may be able to dig out the bits and runtime if you need it asap.

**Comment by epet** [comment]
It depends. If OpenApi v3 is about to be supported and this is resolved, we can wait a sprint or two and fix the client code up ourselves but we are trying to use OOBE to demonstrate to our customers. If it's going to be longer, ya, we might want the dev branch. Please advise.

**Comment by fearthecowboy** [comment]
Yeah, it's going to be a bit longer for OAI3 . lemme go dig it up for you.

**Comment by fearthecowboy** [comment]
Oh. And I should mention that it's also C# only at the moment; the other generators haven't implemented it yet.
